### PR TITLE
Improved parsing functionality

### PR DESCRIPTION
**Pull request by ShaneKent**
Added support for events that might display incorrect data due to parsing that fails. Also improved the parsing for EventID 4624.

### CODE REVIEW

**Comment by ShaneKent** [comment]
@carterv @zjmonroe23 Please review.

File: tests/test_parser.py [hunk 1]
```diff
@@ -364,14 +364,16 @@ def test_parser_id_1074():
...
     record = parser.parse_id_1074(d, rec)  # System    -   Shutdown Initiated
-    data = d["Event"]["EventData"]["Data"]
-    assert(record["account"]) == "SYSTEM"
-    assert(record["details"]) == "Cause: restart"
-    assert(record["computer_name"]) == "LAPTOP-9KUQNI2Q"
-    if data[3]["#text"] == "0x500ff":
-        assert(record["description"]) == "Shutdown Initiated"
-    else:
-        assert(record["description"]) == "ERROR"
+    assert (record) == None  # This is the case because Bruce "tosses" any 1074 records where Data param4 != 0x500ff.
```
Comment (carterv) [inline tests/test_parser.py]: Can you change this test to cover both of the cases for when we parse a record with id 1074 that we do want and when we parse one that we toss?

File: tests/test_parser.py [hunk 2]
```diff
@@ -601,12 +603,21 @@ def test_parser_id_4634():
            }
     record = parser.parse_id_4634(d, rec)
     data = d["Event"]["EventData"]["Data"]
-    assert(record["account"]) == "7"
+    assert (record["account"]) == "sarah"  # This was being incorrectly parsed originally.
```
Comment (carterv) [inline tests/test_parser.py]: I've been wondering about those for a while, glad to see it's been fixed.

File: winlogtimeline/collector/parser.py [hunk 3]
```diff
@@ -70,19 +73,22 @@ def parse_id_12(raw, record):
     record["description"] = "System Start"
 
     if len(data) == 7:
-        # print(data)
-        record["details"] = f'Starting Windows NT version {data[0]["#text"]}. {data[1]["#text"]}. {data[2]["#text"]} ' \
-                            f'at {data[6]["#text"][0:10]} {data[6]["#text"][11:23]} (UTC)'
+        time = get_string(data[6])
```
Comment (carterv) [inline winlogtimeline/collector/parser.py]: The get_string not only fixes inconsistencies with records, but it also makes the code easier to read. Nice job.

File: winlogtimeline/collector/parser.py [hunk 4]
```diff
@@ -305,6 +333,16 @@ def parse_id_6013(raw, record):
...
+
+    uptime = get_string(data[4])
+    timezone = get_string(data[6])
+
+    record["details"] = f'System Uptime: {uptime} seconds. Time Zone Setting: {timezone}.'
 
     return record
+
+
+def get_string(string):
```
Comment (carterv) [inline winlogtimeline/collector/parser.py]: Please add a docstring.

File: tests/test_parser.py (hunk 1 above)
Comment (ShaneKent) [reply tests/test_parser.py]: @zjmonroe23

File: setup.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 
 SETUP_DEPENDENCIES = ['pytest-runner']
 
-TEST_DEPENDENCIES = ['pytest']
+TEST_DEPENDENCIES = ['pytest', 'mock']
```
Comment (zjmonroe23) [inline setup.py]: Amen.

**Pull request merged**